### ISSUE TITLE
Initiating User Registration via OpenID Connect

### ISSUE THREAD

**Issue by jaxoncreed** [solid/authentication-panel#46]
We might want to consider integrating this draft recommendation in the solid auth spec: https://openid.net/specs/openid-connect-prompt-create-1_0.html as it would be especially useful for Solid.

Traditional OIDC presupposes that a user is already registered with an IDP before the login process begins. However, with solid, a user might want to use a solid-compatible application without ever having registered with a solid-compatible IDP.

It would be useful to allow app developers to provide a "register" button that initiates the login process beginning with a registration flow rather than a login flow.

**Comment by acoburn** [comment]
Just want to note here that this could be an interesting _optional_ feature. Requiring support, however, would be hugely problematic for anyone implementing an identity broker, which is important for integrating Solid with OAuth2, SAML or traditional OIDC environments -- or, in fact, with any IdP that doesn't currently support Solid

**Comment by elf-pavlik** [comment]
I think this shares some issues with #5 and also tries to deal with User Lifecycle. I see IdP as much higher priority role (especially if it hosts WebID) than any single client application. For that reason I would really hesitate to encourage applications to suggest specific IdP to the user. How about we try to explore those on boarding scenario in https://github.com/solid/user-lifecycle-panel and bring back some specific authN related requirements afterwards?

In general application should have less expectations on longevity, IdP should ensure certain longevity especially if they host user's WebID.

**Comment by michielbdejong** [comment]
I think even if the IDP honours this request to prompt-to-create, it will still have a "or do you already have an account?" button or link.
And even if the IDP does not understand this optional request to prompt-to-create, its default login screen will still have a "or register here" button or link.
So yes, useful feature,
And yes, also degrades very gracefully if it's optional.

**Comment by michielbdejong** [comment]
See also gitter discussion at https://gitter.im/solid/app-development?at=5f77ac77f9cfbe19f935cf10

**Comment by woutermont** [comment]
Of course, current classic OIDC providers could perfectly implement the create prompt, but have not (yet). 

I fail to see, however, how that prevents Solid-OIDC from adopting it. After all, the current draft is already so complex that it is impossible to make most popular OIDC providers compliant without some serious backend development:

- Some providers don't support DPoP (which is also still a draft btw).
- A number of clients do not allow to customize the OIDC Discovery document with a solid support claim.
- No provider currently supports checking ClientID documents.

Yet in none of these cases, such lack of current support has prevented Solid-OIDC from making it mandatory.

Moreover, we are indeed actively trying to implement this functionality in client projects and our own products! That is exactly why I am asking the panel and community to keep brainstorming about this. The solution does not have to be a create prompt, but the demand is definitely there.

**Comment by acoburn** [comment]
> After all, the current draft is already so complex that it is impossible to make most popular OIDC providers compliant without some serious backend development

By that argument, adding another requirement to this specification makes very little sense.

> No provider currently supports checking ClientID documents

I can think of two that currently support this.
